security/pkg/platform/client.go

```diff
@@ -37,20 +39,30 @@ type Client interface {
 	GetCredentialType() string
 }
 
+const (
+	onPrem      = "onprem"
```

Comment:
You wont run into package name confliction issue if you make these names more specific.

`gcpVM`, `awsVM`, etc.

I also suggest you to add another typedef for these platforms. `typedef VMType string`